lib/spack/spack/cmd/style.py

```diff
@@ -239,7 +252,7 @@ def run_mypy(file_list, args):
         return 1
 
     print_tool_header("mypy")
-    mpy_args = ["--package", "spack", "--package", "llnl"]
+    mpy_args = ["--package", "spack", "--package", "llnl", "--show-error-codes"]
```

Comment:
Not related to this change, but makes it significantly easier to know how to disable a mypy error.

Reply:
Agreed, should have done this from the start.

Reply:
We should move these to `.mypy.ini`. In fact, we should combine `.mypy.ini` into `pyproject.toml` so that we have one fewer top-level config file.

Reply:
It looks like `pyproject.toml` support was added in `mypy>=0.900` so we should update the version requirement too.